watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,460 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+from pyomo.environ import (
+    ConcreteModel,
+    Var,
+    value,
+    Constraint,
+    Objective,
+    TransformationFactory,
+    assert_optimal_termination,
+    units as pyunits,
+    NonNegativeReals,
+)
+from pyomo.network import Arc
+
+from idaes.core import FlowsheetBlock, UnitModelCostingBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.initialization import propagate_state
+import idaes.core.util.model_statistics as mstat
+from idaes.models.unit_models import Feed, Product, Separator, Mixer
+from watertap.unit_models.pressure_changer import Pump
+from idaes.models.unit_models.mixer import MixingType, MomentumMixingType
+import pandas as pd
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslogger
+from watertap.unit_models.electrodialysis_1D import (
+    ElectricalOperationMode,
+    PressureDropMethod,
+    FrictionFactorMethod,
+    HydraulicDiameterMethod,
+    LimitingCurrentDensityMethod,
+)
+from watertap.unit_models.electrodialysis_1D import Electrodialysis1D
+from watertap.costing.watertap_costing_package import WaterTAPCosting
+from watertap.property_models.multicomp_aq_sol_prop_pack import MCASParameterBlock
+
+__author__ = "Xiangyu Bi"
+_log = idaeslogger.getLogger(__name__)
+
+
```

Comment:
Nice to have: a doc string telling the user what this flowsheet is intended to represent.